lib/trino-filesystem-azure/src/main/java/io/trino/filesystem/azure/AzureFileSystem.java

```diff
@@ -0,0 +1,342 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.azure;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobContainerClientBuilder;
+import com.azure.storage.blob.models.AccountKind;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.models.ListBlobsOptions;
+import com.azure.storage.blob.models.StorageAccountInfo;
+import com.azure.storage.common.Utility;
+import com.azure.storage.file.datalake.DataLakeDirectoryClient;
+import com.azure.storage.file.datalake.DataLakeFileClient;
+import com.azure.storage.file.datalake.DataLakeFileSystemClient;
+import com.azure.storage.file.datalake.DataLakeServiceClient;
+import com.azure.storage.file.datalake.DataLakeServiceClientBuilder;
+import com.azure.storage.file.datalake.models.DataLakeRequestConditions;
+import com.azure.storage.file.datalake.models.ListPathsOptions;
+import com.azure.storage.file.datalake.models.PathItem;
+import com.azure.storage.file.datalake.options.DataLakePathDeleteOptions;
+import io.airlift.units.DataSize;
+import io.trino.filesystem.FileIterator;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.TrinoOutputFile;
+
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.OptionalLong;
+
+import static com.azure.storage.common.implementation.Constants.HeaderConstants.ETAG_WILDCARD;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.filesystem.azure.AzureUri.toDirectoryUri;
+import static io.trino.filesystem.azure.AzureUri.toFileUri;
+import static io.trino.filesystem.azure.AzureUtils.handleAzureException;
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Predicate.not;
+
+public class AzureFileSystem
+        implements TrinoFileSystem
+{
+    private final HttpClient httpClient;
+    private final AzureAuth azureAuth;
+    private final int readBlockSizeBytes;
+    private final long writeBlockSizeBytes;
+    private final int maxWriteConcurrency;
+    private final long maxSingleUploadSizeBytes;
+
+    public AzureFileSystem(HttpClient httpClient, AzureAuth azureAuth, AzureFileSystemConfig config)
```

Comment:
Let's remove this and have the tests use `AzureFileSystemFactory`. In normal usage this won't be constructed with config objects, so seems better not to do it from tests.